cudaungappedextender/include/claraparabricks/genomeworks/cudaungappedextender/cudaungappedextender.hpp

```diff
@@ -0,0 +1,92 @@
+/*
+* Copyright 2020 NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#pragma once
+
+namespace claraparabricks
+{
+namespace genomeworks
+{
+namespace cudaungappedextender
+{
+
+struct Anchor
+{
+    /// position of first sketch element in query_read_id_
+    position_in_read_t query_position_in_read_;
+    /// position of second sketch element in target_read_id_
+    position_in_read_t target_position_in_read_;
+};
+
+struct ScoredSegment
+{
+    /// Anchor for the segment
+    Anchor anchor;
+    /// length of the segment
+    int32_t len;
+    /// score of the segment
+    int32_t score;
+};
+
+/// CUDA Ungapped Extension object
+class UngappedExtender
+{
+public:
+    /// \brief Destructor for UngappedExtender.
+    ~UngappedExtender() = default;
```

Comment:
Since this is a concrete class and no other special functions are defined, there is no need to explicitly define the destructor as default. Let the compiler do it for you implicitly.